modules/add-contextual-data/context-info-db.c

```diff
@@ -260,19 +260,37 @@ context_info_db_get_selectors(ContextInfoDB *self)
   return g_hash_table_get_keys(self->index);
 }
 
+static void
+_truncate_eol(gchar *line, gsize line_len)
+{
+  if (line[line_len - 2] == '\r')
+    line[line_len - 2] = '\0';
+  else if (line[line_len - 1] == '\n')
+    line[line_len - 1] = '\0';
```

Comment:
* what happens when `line_len < 2`?
* what happens when this is a last line in a file without a new line and it contains a stray `'\r'`, for example `"xy\rz"`?
* although it is unused as of now, I'd probably adjust `line_buf_len` as well